client-api/src/test/java/org/sonarsource/sonarlint/core/client/api/util/FileUtilsTest.java

```diff
@@ -48,6 +54,32 @@ public void moveDir_should_rename_a_dir() throws IOException {
     assertThat(newDir.isDirectory()).isTrue();
   }
 
+  @Test
+  public void moveDir_non_atomic() throws IOException {
+    File oldDir = temporaryFolder.newFolder();
+    File foo = new File(oldDir, "foo");
+    foo.mkdirs();
+    new File(foo, "foo.txt").createNewFile();
+
+    File newDir = temporaryFolder.newFolder();
+
+    Path destPath = newDir.toPath();
+
+    Path destMock = mock(Path.class, RETURNS_DEEP_STUBS);
+    // Make the dest path look like it doesn't belong to the same FileSystem
+    when(destMock.getFileSystem().provider()).thenReturn(null);
+    when(destMock.resolve(any(Path.class))).thenAnswer(new Answer<Path>() {
+      @Override
+      public Path answer(InvocationOnMock invocation) throws Throwable {
+        return destPath.resolve(invocation.<Path>getArgument(0));
+      };
+    });
+
+    FileUtils.moveDir(oldDir.toPath(), destMock);
+    assertThat(oldDir.isDirectory()).isFalse();
```

Comment:
Instead of asserting it's not a directory, it would be stronger and more to the point to assert it doesn't exist.